### DISCUSSION TITLE
xit! for vscode

### DISCUSSION THREAD

**Discussion by tscpp** [jotaen/xit#20] [Show & tell]
I made an extension for vscode to provide xit language support.

https://marketplace.visualstudio.com/items?itemName=tscpp.xit
https://github.com/tscpp/xit-vscode

**Comment by jotaen** [comment]
Cool that you made this! Let me know if (or when) I should link this [from the website](https://xit.jotaen.net/).

Unfortunately, there is no good way of testing whether a syntax highlighting conforms to the (current version of the) spec. For my own [Sublime Plugin](https://packagecontrol.io/packages/xit!), I use a [test file for spot-checking](https://github.com/jotaen/xit-sublime/blob/main/resources/test.xit), which I’ve posted a screenshot of below. I’m just sharing this in case it might be helpful for you, you shouldn’t feel obligated to conform to it in any way.

There is already [a discussion regarding the ability to test tools](https://github.com/jotaen/xit/discussions/6) against the spec. It would probably be useful to drive that forward. Just as an aside, the Sublime plugin is also not perfect, as it for example highlights all occurrences of a due date in a description, even though [only the first one should be recognised](https://github.com/jotaen/xit/blob/main/Specification.md#due-date).

<img width="371" alt="syntax-check" src="https://user-images.githubusercontent.com/3618384/161397282-e2b4d3eb-9b38-46a6-9b22-4ca5801450e7.png">

**Reply by tscpp** [reply]
Link the extension whenever. The extension is not polished and have a few [issues](https://github.com/tscpp/xit-vscode). I'll maintain it and make sure it's up to date. I use todo lists on a daily basis and switched to using xit.

I don't feel the need for testing the extension because how simple it is. I have, like you, a sample file which I use for testing. Your sample file looks very well organized. I will switch to it when making further improvements on the extension.

**Reply by tscpp** [reply]
Is it okay if I use the icon on your website as the icon of my extension and possibly file icon? I can link copyright or license if needed.

**Reply by jotaen** [reply]
I’ve added a link to your extension to the website.

> Is it okay if I use the icon on your website as the icon of my extension and possibly file icon?

Thanks for checking that. I’d like to keep the logo artwork unique to the website, so that it’s clear that every tool stands on its own, and is recognised as such. (That also applies to my own Sublime Package, btw.)

However, the `[x]it!` name is subject to [the same license as the spec itself (CC0/OWFa)](https://github.com/jotaen/xit/blob/main/Specification.md#license), and for the `[x]` symbol I doubt that it’s eligible for copyright at all. So you are free to create your own icon or logo artwork based on them.

**Comment by jotaen** [comment]
Just fyi, for improved test data: https://github.com/jotaen/xit/issues/22
